src/app/components/home/home.component.ts

```diff
@@ -8,4 +19,57 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class HomeComponent {
+  readonly categoriesList$: Observable<CategoryModel[]> = this._categoriesService.getAllCategories();
+  readonly storesList$: Observable<StoreModel[]> = this._storesService.getAllStores();
+  readonly storesTagsList$: Observable<StoreTagModel[]> = this._storesService.getAllStoreTags()
+  readonly productsList$: Observable<ProductModel[]> = this._productsService.getAllProducts();
+  readonly categoryWithProductsIds$: Observable<string[]> = of(['5', '2']);
+  readonly storesWithTagsList$: Observable<StoreWithTagQueryModel[]> = combineLatest([
+    this.storesList$,
+    this.storesTagsList$,
+  ]).pipe(
+    map(([stores, tags]) => this._mapToStoreWithTagsQuery(stores, tags))
+  )
+
+  readonly categoriesWithProductsList$: Observable<CategoryWithProductQueryModel[]> = combineLatest([
+    this.productsList$,
+    this.categoriesList$,
+    this.categoryWithProductsIds$
+  ]).pipe(
+    map(([products, categories, ids]) => this._mapToCategoryWithProductQuery(products, categories, ids))
+  )
+
+
+  private _mapToStoreWithTagsQuery(stores: StoreModel[], tags: StoreTagModel[]): StoreWithTagQueryModel[] {
+    const tagsMap = tags.reduce((a, c) => ({ ...a, [c.id]: c }), {} as Record<string, StoreTagModel>)
+
+    return stores.map(store => ({
+      id: store.id,
+      name: store.name,
+      logoUrl: store.logoUrl,
+      distanceInKilometers: +(store.distanceInMeters / 1000).toFixed(1),
+      storeTags: (store.tagIds ?? []).map(id => tagsMap[id]?.name)
+    }))
+  }
+
+  constructor(private _categoriesService: CategoriesService, private _storesService: StoresService, private _productsService: ProductsService) {
+  }
+
+  private _mapToCategoryWithProductQuery(products: ProductModel[], categories: CategoryModel[], ids: string[]) : CategoryWithProductQueryModel[] {
+    const categoryMap = categories.reduce((a, c) => ({ ...a, [c.id]: c}), {} as Record<string, CategoryModel>)
+    return ids.map(id => ({
+      categoryId: id,
+      categoryName: categoryMap[id]?.name,
+      products: products.filter(product => product.categoryId === id)
```

Comment:
Here is a good place to use reduce method because you used filter and map methods.